src/Common.fs

```diff
@@ -15,7 +15,7 @@ type SqlValue =
     | Int of int
     | Long of int64
     | String of string
-    | Date of DateTime
+    | Date of Choice<DateTime, DateOnly>
```

Comment:
Not sure about this breaking change. Technically, these are both valid Npgsql representations of Postgres' `date`, so from a modelling standpoint it makes sense that it remains a single case.